### PR TITLE
Allow regexp for a allowed_request_origins array

### PR DESCRIPTION
**Pull request by adamliesko**
Adding the feature of additional regexp check for `allowed_origin_requests`instead of only strict equal comparation.

Fixes #130

### CODE REVIEW

File: lib/action_cable/connection/base.rb [hunk 1]
```diff
@@ -180,6 +180,11 @@ def allow_request_origin?
           end
         end
 
+        def allowed_origins_match? origin
+          allowed_origins = Array(server.config.allowed_request_origins)
+          allowed_origins.any? { |allowed_origin| allowed_origin.is_a?(Regexp) ? allowed_origin =~ origin : allowed_origin == origin }
```
Comment (lsylvester) [inline lib/action_cable/connection/base.rb]: You could use the `===` operator to match both strings and regexes, so the class check wouldn't be required.

```
allowed_origins.any? { |allowed_origin| allowed_origin === origin }
```

File: lib/action_cable/connection/base.rb [hunk 2]
```diff
@@ -180,6 +180,11 @@ def allow_request_origin?
           end
         end
 
+        def allowed_origins_match? origin
```
Comment (dhh) [inline lib/action_cable/connection/base.rb]: Style: Use parenthesis in method definition signature.

File: lib/action_cable/connection/base.rb (hunk 1 above)
Comment (dhh) [inline lib/action_cable/connection/base.rb]: Actually, given this, we don't need the extra method. We can just inline this === check like we had the original include call.

Comment (adamliesko) [inline lib/action_cable/connection/base.rb]: Sure, that makes sense. Thanks.

**Comment by dhh** [comment]
@adamliesko Do you want to wrap up these changes? Then we can merge.

**Comment by adamliesko** [comment]
@dhh I have updated the `README.md` and rebased the branch anything else left to do?

File: lib/action_cable/connection/base.rb (hunk 1 above)
Comment (lsylvester) [inline lib/action_cable/connection/base.rb]: @adamliesko I don't think this method is used anymore and can be removed.

**Comment by dhh** [comment]
👏

**Pull request merged**